File: app/src/main/java/com/android/exercise/ui/activity/player/IjkPlayerActivity.java
package com.android.exercise.ui.activity.player;

import android.os.Bundle;
import android.view.View;

import androidx.annotation.Nullable;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.widget.Toolbar;

import com.android.exercise.R;
import com.android.exercise.base.BaseActivity;
import com.android.exercise.base.toolbar.ToolBarCommonHolder;
import com.android.exercise.ui.widget.player.IjkPlayerView;

/**
 * ijkPlayer
 * Created by wangzhen on 2020-01-15.
 */
public class IjkPlayerActivity extends BaseActivity {
    private static final String VIDEO_URL = "http://10.100.119.192:8080/wangzhen/video/beauty.mp4";
    private IjkPlayerView mPlayerView;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_ijk_player);
        mPlayerView = (IjkPlayerView) findViewById(R.id.player_view);
    }

    @Override
    protected void onSetupToolbar(Toolbar toolbar, ActionBar actionBar) {
        new ToolBarCommonHolder(this, toolbar, "ijkPlayer");
    }

    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.btn_play:
                mPlayerView.play(VIDEO_URL);
            case R.id.btn_pause:
                mPlayerView.pause();
            case R.id.btn_resume:
                mPlayerView.resume();
            case R.id.btn_stop:
                mPlayerView.stop();
            case R.id.btn_reset:
                mPlayerView.reset();
        }
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mPlayerView.release();
    }
}

File: app/src/main/java/com/android/exercise/ui/activity/player/IjkPlayerActivity.kt
package com.android.exercise.ui.activity.player

import android.os.Bundle
import android.support.v7.app.ActionBar
import android.support.v7.widget.Toolbar
import android.view.View
import com.android.exercise.R
import com.android.exercise.base.BaseActivity
import com.android.exercise.base.toolbar.ToolBarCommonHolder
import com.android.exercise.ui.widget.player.IjkPlayerView

/**
 * ijkPlayer
 * Created by wangzhen on 2020-01-15.
 */
class IjkPlayerActivity : BaseActivity() {
    private val VIDEO_URL = "http://10.100.119.192:8080/wangzhen/video/beauty.mp4"
    private lateinit var mPlayerView: IjkPlayerView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_ijk_player)
        mPlayerView = findViewById(R.id.player_view) as IjkPlayerView
    }

    override fun onSetupToolbar(toolbar: Toolbar?, actionBar: ActionBar?) {
        ToolBarCommonHolder(this, toolbar, "ijkPlayer")
    }

    fun onClick(view: View) {
        when (view.id) {
            R.id.btn_play -> {
                mPlayerView.play(VIDEO_URL)
            }
            R.id.btn_pause -> {
                mPlayerView.pause()
            }
            R.id.btn_resume -> {
                mPlayerView.resume()
            }
            R.id.btn_stop -> {
                mPlayerView.stop()
            }
            R.id.btn_reset -> {
                mPlayerView.reset()
            }
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        mPlayerView?.release()
    }
}
